### PR TITLE
Homebrew Update for Mac

### PR DESCRIPTION
**Pull request by abhishek-galoda**
3.0.0 - Homebrew supports Mac M1

The command which worked on my Mac M1 was  "arch -arm64 brew install dapr/tap/dapr-cli"

Here is the output 

arch -arm64 brew install dapr/tap/dapr-cli
==> Downloading https://ghcr.io/v2/homebrew/core/go/manifests/1.17.2
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/go/blobs/sha256:b0a1686c5c5ba668e78a97d66567e9d007f7d2a0c9b1a53e79841e3736d729e6
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:b0a1686c5c5ba668e78a97d66567e9d007f7d2a0c9b1a53e79841e3736d729e6?se=2021-10-20T18%3A00%3A00Z&sig=yRfOG5pLjci12lOR2LOurCSGZjgWHgTKSV00%2B8Ys2PI%3D&sp=r&spr=https&sr=b&sv=2019-12-12
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/make/manifests/4.3-1
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/make/blobs/sha256:eab3fbc3688aecec0fe90b8d0fe3cb7beb84ed773ba0411fc2f855c66deaf882
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:eab3fbc3688aecec0fe90b8d0fe3cb7beb84ed773ba0411fc2f855c66deaf882?se=2021-10-20T18%3A05%3A00Z&sig=YsB%2FxVvM7Q%2BjIVUXWwBGgGYPo%2F7lYMM5RohWXPKe3gY%3D&sp=r&spr=https&sr=b&sv=2019-12-12
######################################################################## 100.0%
==> Downloading https://github.com/dapr/cli/archive/v1.4.0.tar.gz
==> Downloading from https://codeload.github.com/dapr/cli/tar.gz/v1.4.0
#-#O=#  #                                                                     
==> Installing dapr-cli from dapr/tap
Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 13.0.

==> Installing dependencies for dapr/tap/dapr-cli: go and make
==> Installing dapr/tap/dapr-cli dependency: go
==> Pouring go--1.17.2.arm64_big_sur.bottle.tar.gz
🍺  /opt/homebrew/Cellar/go/1.17.2: 10,814 files, 564.3MB
==> Installing dapr/tap/dapr-cli dependency: make
==> Pouring make--4.3.arm64_big_sur.bottle.1.tar.gz
🍺  /opt/homebrew/Cellar/make/4.3: 15 files, 1MB
==> Installing dapr/tap/dapr-cli
==> make REL_VERSION=1.4.0
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - dist/darwin_amd64/release/dapr

The brew formula needs to be corrected however.

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### CODE REVIEW

File: daprdocs/content/en/getting-started/install-dapr-cli.md [hunk 1]
```diff
@@ -56,10 +56,10 @@ brew install dapr/tap/dapr-cli
 ```
 
 #### Note for M1 Macs
-For M1 Macs, homebrew is not supported. You will need to use the dapr install script and have the rosetta amd64 compatibility layer installed. If you do not have it installed already, you can run the following:
+For M1 Macs, only homebrew 3.0+ versions are supported. Please update Homebrew to 3.0.0+ and use below command
```
Comment (paulyuk) [inline daprdocs/content/en/getting-started/install-dapr-cli.md]:
```suggestion
For M1 Macs, only Homebrew 3.0 and higher versions are supported. Please update Homebrew to 3.0.0 or higher and the command below:
```

Comment (paulyuk) [reply daprdocs/content/en/getting-started/install-dapr-cli.md]: Updating text to be a bit more consistent with styles.  Note brew self updates but I think it's ok to say please update.

**Review by paulyuk — CHANGES_REQUESTED** [review]
Thank you for contributing this.  

Issue tho - On my MBP M1 the suggested command fails always.  I also tried fully uninstalling Dapr CLI and runtime first.  What is most suspicious is `No such file or directory - dist/darwin_amd64/release/dapr`. Do we need to have an ARM build in an ARM folder or are we relying on interop?

My error:
 paulyuk@Pauls-MacBook-Pro  ~/src/dotnetapi  which dapr
dapr not found
 ✘ paulyuk@Pauls-MacBook-Pro  ~/src/dotnetapi  arch -arm64 brew install dapr/tap/dapr-cli
==> Downloading https://github.com/dapr/cli/archive/v1.4.0.tar.gz
Already downloaded: /Users/paulyuk/Library/Caches/Homebrew/downloads/7337acd8de7340b67e497e4a7e9034f6b493ee601918b62ece4ef0927cae64ad--cli-1.4.0.tar.gz
==> Installing dapr-cli from dapr/tap
==> make REL_VERSION=1.4.0
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - dist/darwin_amd64/release/dapr

**Comment by willtsai** [comment]
Hi @abhishek-galoda - thank you for looking into this. Like @paulyuk I'm also not able to install Dapr using your proposed `brew` command with the same error message:
```
Error: An exception occurred within a child process:
Errno::ENOENT: No such file or directory - dist/darwin_amd64/release/dapr
```

It seems like you're running into the same issue and you also mention "The brew formula needs to be corrected however." -- can you elaborate on this more please? Are you saying that the Dapr installation script (or brew formula) needs to be updated to accommodate the M1 Macs style of installation? If so, we need to create an Issue+PR against the https://github.com/dapr/homebrew-tap repo before we can properly test your instructions in this docs PR.

**Comment by willtsai** [comment]
@abhishek-galoda -- following up here to see if you've been able to look into my questions above.

Thanks,
-Will

File: daprdocs/content/en/getting-started/install-dapr-cli.md (hunk 1 above)
Comment (berndverst) [reply daprdocs/content/en/getting-started/install-dapr-cli.md]: Also add:

When installing from terminal native ARM64 binaries are downloaded where available. For older releases AMD64 binaries are downloaded which must be run with Rosetta2 emulation enabled. To install Rosetta emulation, run `softwareupdate --install-rosetta`.

Comment (willtsai) [reply daprdocs/content/en/getting-started/install-dapr-cli.md]: @berndverst - for the paragraph you recommended above, it should be added to the `### Install from Terminal` section of this doc, correct?

Comment (willtsai) [inline daprdocs/content/en/getting-started/install-dapr-cli.md]: Minor edit to @paulyuk's suggestion to rename M1 to ARM64 per offline conversation with @berndverst 
```suggestion
For ARM64 Macs, only Homebrew 3.0 and higher versions are supported. Please update Homebrew to 3.0.0 or higher and the command below:
```

File: daprdocs/content/en/getting-started/install-dapr-cli.md [hunk 2]
```diff
@@ -56,10 +56,10 @@ brew install dapr/tap/dapr-cli
 ```
 
 #### Note for M1 Macs
```
Comment (willtsai) [inline daprdocs/content/en/getting-started/install-dapr-cli.md]:
```suggestion
#### Note for ARM64 Macs
```

**Review by willtsai — CHANGES_REQUESTED** [review]

**Review by greenie-msft — APPROVED** [review]

**Review by willtsai — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

File: daprdocs/content/en/getting-started/install-dapr-cli.md [hunk 3]
```diff
@@ -48,18 +48,22 @@ This command installs the latest darwin Dapr CLI to `/usr/local/bin`:
 ```bash
 curl -fsSL https://raw.githubusercontent.com/dapr/cli/master/install/install.sh | /bin/bash
 ```
+When installing from the terminal, native ARM64 binaries are downloaded when available. For older releases, AMD64 binaries are downloaded, which must be run with Rosetta2 emulation enabled. To install Rosetta emulation:
+```bash
+run softwareupdate --install-rosetta
```
Comment (willtsai) [inline daprdocs/content/en/getting-started/install-dapr-cli.md]:
```suggestion
#### Note for ARM64 Macs
Support for ARM64 Macs is available, but currently in the Preview phase. When installing from the terminal, native ARM64 binaries are downloaded when available. For older releases, AMD64 binaries are downloaded, which must be run with Rosetta2 emulation enabled. To install Rosetta emulation:
```bash
softwareupdate --install-rosetta
```

File: daprdocs/content/en/getting-started/install-dapr-cli.md [hunk 4]
```diff
@@ -48,18 +48,22 @@ This command installs the latest darwin Dapr CLI to `/usr/local/bin`:
...
 ### Install from Homebrew
 You can install via [Homebrew](https://brew.sh):
 ```bash
 brew install dapr/tap/dapr-cli
 ```
 
-#### Note for M1 Macs
-For M1 Macs, homebrew is not supported. You will need to use the dapr install script and have the rosetta amd64 compatibility layer installed. If you do not have it installed already, you can run the following:
+#### Note for ARM64 Macs
+Support for ARM64 Macs is available, but currently in the experimental phase. For ARM64 Macs, only Homebrew 3.0 and higher versions are supported. Please update Homebrew to 3.0.0 or higher and the command below:
```
Comment (willtsai) [inline daprdocs/content/en/getting-started/install-dapr-cli.md]:
```suggestion
For ARM64 Macs, only Homebrew 3.0 and higher versions are supported. Please update Homebrew to 3.0.0 or higher and the command below:
```

**Review by willtsai — CHANGES_REQUESTED** [review]
a few more edits.

**Review by greenie-msft — APPROVED** [review]

**Comment by willtsai** [comment]
> [CLA assistant check](https://cla.opensource.microsoft.com/dapr/docs?pullRequest=1871) Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/dapr/docs?pullRequest=1871) before we can accept your contribution.❌ abhishek-galoda [sign now](https://cla.opensource.microsoft.com/dapr/docs?pullRequest=1871)You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/dapr/docs?pullRequest=1871) it.

@abhishek-galoda - can you please sign this Contributor License Agreement ASAP so that we may merge in your contributions?

**Comment by willtsai** [comment]
Since we need these changes merged in today for the v1.5 release and this PR is blocked by the CLA, I've duplicated the changes into https://github.com/dapr/docs/pull/1935 which will be merged today. We've made sure to include @abhishek-galoda in the v1.5 release notes acknowledgements though.

We should close this PR now without merging, thanks!

**Pull request closed without merging**